src/ui/public/vislib/lib/tilemap_settings.js

```diff
@@ -0,0 +1,135 @@
+import uiModules from 'ui/modules';
+import _ from 'lodash';
+import marked from 'marked';
+import url from 'url';
+
+marked.setOptions({
+  gfm: true, // Github-flavored markdown
+  sanitize: true // Sanitize HTML tags
+});
+
+
+uiModules.get('kibana')
+  .service('tilemapSettings', function ($http, mapsConfig, $sanitize) {
+
+    const attributionFromConfig = $sanitize(marked(mapsConfig.deprecated.config.options.attribution));
+    const optionsFromConfig = _.assign({}, mapsConfig.deprecated.config.options, {attribution: attributionFromConfig});
+
+    /**
+     * Object to read out the map-service configuration
+     *
+     * @type {Object}
+     */
+    const mapSettings = {
+
+      _queryParams: {license: ""},//this curheprently a mandatory parameter. Remove when no longer the case.
+      _settingsInitialized: false,
+
+      //initialize settings with the default of the configuration
+      _url: mapsConfig.deprecated.config.url,
+      _options: optionsFromConfig,
+
+      /**
+       * Must be called before getUrl/getOptions can be called.
+       */
+      whenSettingsReady: async function () {
```

Comment:
After reading on it became clear that this class should probably support invalidating the loaded (like when the query params are modified). Perhaps an implementation like this is more appropriate:

```js
class MapSettings {
  ensureSettingsReady = once(this._loadSettings);

  invalidateSettings() {
    // override the ensureSettingsReady function so that
    // subsequent calls trigger a new call to _loadSettings()
    this.ensureSettingsReady = once(this._loadSettings);
  }

  _loadSettings() {
    if (!this._loadSettingsPromise) {
      this._loadSettingsPromise = attempt(() => {
        // load the settings
      })
    }

    return this._loadSettingsPromise;
  }
}
```

Then when `addQueryParams()` is called it can invalidate the settings object and force `_loadSettings()` to run on the next route change.